packages/dd-trace/src/plugins/ci_plugin.js

```diff
@@ -0,0 +1,136 @@
+const { channel } = require('diagnostics_channel')
+
+const {
+  getTestEnvironmentMetadata,
+  getCodeOwnersFileEntries,
+  getTestParentSpan,
+  getTestCommonTags,
+  getCodeOwnersForFilename,
+  TEST_CODE_OWNERS,
+  CI_APP_ORIGIN
+} = require('./util/test')
+const { getItrConfiguration } = require('../ci-visibility/intelligent-test-runner/get-itr-configuration')
+const { getSkippableSuites } = require('../ci-visibility/intelligent-test-runner/get-skippable-suites')
+const { COMPONENT } = require('../constants')
+
+const Plugin = require('./plugin')
+
+module.exports = class CiPlugin extends Plugin {
+  constructor (...args) {
+    super(...args)
+
+    const gitMetadataUploadFinishCh = channel('ci:git-metadata-upload:finish')
+    // `gitMetadataPromise` is used to wait until git metadata is uploaded to
+    // proceed with calculating the suites to skip
+    // TODO: add timeout after which the promise is resolved
+    const gitMetadataPromise = new Promise(resolve => {
+      gitMetadataUploadFinishCh.subscribe(err => {
+        resolve(err)
+      })
+    })
+
+    this.testEnvironmentMetadata = getTestEnvironmentMetadata(this.constructor.name, this.config)
```

Comment:
Using `this.config` is not safe at that point until `configure` has been called. It's even a bit worrying that the tests are passing with this change. I'd move that code to `configure` instead.